cnd/src/swap_protocols/rfc003/events/ledger_event_futures.rs

```diff
@@ -32,16 +31,14 @@ impl<L: Ledger, A: Asset> LedgerEventFutures<L, A> {
             htlc_redeemed_or_refunded: None,
         }
     }
-}
 
-impl<L: Ledger, A: Asset> LedgerEvents<L, A> for LedgerEventFutures<L, A> {
-    fn htlc_deployed(&mut self, htlc_params: HtlcParams<L, A>) -> &mut DeployedFuture<L> {
+    pub fn htlc_deployed(&mut self, htlc_params: HtlcParams<L, A>) -> &mut DeployedFuture<L> {
         let htlc_events = &self.htlc_events;
         self.htlc_deployed
             .get_or_insert_with(move || htlc_events.htlc_deployed(htlc_params))
     }
 
-    fn htlc_funded(
+    pub fn htlc_funded(
```

Comment:
Instead of making this pub, you could just move the struct into the `state_machine` module because it is now an implementation detail :)

Reply:
+ rename it to `HtlcEventFutures` then the name makes more sense when looked at in combination with `HtlcEvents`.